scripts/adminrouter-agent-setup.ps1

```diff
@@ -0,0 +1,125 @@
+        [Parameter(Mandatory=$true)][string]$Name,
+        [Parameter(Mandatory=$true)][string]$MSISourceUri,
+        [Parameter(Mandatory=$true)][string]$DownloadDestFile,
+        [Parameter(Mandatory=$true)][string]$Logfile
+        )
+
+    Write-Output "Download and install $Name module"
+    Invoke-WebRequest -Uri $MSISourceUri -OutFile $DownloadDestFile
+    $arguments= ' /qn /l*v ' + $Logfile
+
+    Start-Process `
+         -file  $DownloadDestFile `
+         -arg $arguments `
+         -passthru | wait-process
+    if ($? -ne $true)
+    {
+        Write-Host "Could not install $Name module"
+    }
+
+}
+
+Function Install-URLRewriteAndARRModules{
+
+    $downloadDir = Join-Path $env:SystemDrive "dcos-download/"
+
+    # URL Rewrite
+    $UrlRewrite = "IIS URL Rewrite Module 2"
+    $UrlRewritelogfile = "rewrite.log"
+    [Uri]    $URLRewrite_Plugin_Uri  = 'http://download.microsoft.com/download/D/D/E/DDE57C26-C62C-4C59-A1BB-31D58B36ADA2/rewrite_amd64_en-US.msi'
+    [string] $URLRewrite_Plugin_File = $downloadDir + "rewrite_amd64_en-US.msi"
+
+    # Application Request Routing: AAR
+    $ARR = "Microsoft Application Request Routing 3.0"
+    $ARRlogfile = "apprequestrouting.log"
+    [Uri]    $ARR_Plugin_Uri = "https://download.microsoft.com/download/E/9/8/E9849D6A-020E-47E4-9FD0-A023E99B54EB/requestRouter_amd64.msi"
+    [string] $ARR_Plugin_File = $downloadDir + "requestRouter_amd64.msi"
+
+	New-Item -Path $downloadDir -ItemType directory -Force
+
+    Install-AdminModule -Name $UrlRewrite -MSISourceUri $URLRewrite_Plugin_Uri -DownloadDestFile $URLRewrite_Plugin_File -Logfile $ARRlogfile 
+    Install-AdminModule -Name $ARR -MSISourceUri $ARR_Plugin_Uri -DownloadDestFile $ARR_Plugin_File -Logfile $UrlRewritelogfile
+
+}
+
+function Install-IISAndTools{
+	$RequiredComponents = @("Web-Server","Web-Mgmt-Tools","Web-Request-Monitor", "Web-Http-Tracing")
+	foreach ($component in $RequiredComponents) {
+		$feature = get-windowsfeature | where { $_.Name -like $element }
+		if ($feature.InstallState -Eq "Installed"){
+		        Write-Host " 	$component was already installed"
+        } else {
+	  	    Install-WindowsFeature $component
+		}
+	}
+}
+
+function Open-AdminRouterPort {
+    $AdminRouterPort = 61001
+    Open-WindowsFirewallRule -Name "Allow inbound TCP Port $AdminRouterPort for AdminRouter" -Direction "Inbound" -LocalPort $AdminRouterPort -Protocol "TCP"
+}
+
+function Setup-AdminRouter(
+           [string[]]$MasterAddress,
+           [string]$AgentPrivateIP) {
+
+    #  Install IIS and tools
+    Install-IISAndTools
+    import-module WebAdministration
+
+ 	#  Install URL Rewrite and Application Request Routing modules
+    Install-URLRewriteAndARRModules
+
+    # Setup the adminrouter server      
+	Write-Host "Setting up the AdminRouter server"
+	$adminRouterPath="IIS:\AppPools\AdminRouter"
+	$existed = Test-Path -Path $adminRouterPath
+	if (! $existed -eq $True)
+	{
+		Write-Host "creating IIS:\AppPools\AdminRouter"
```

Comment:
I'd suggesting using `Write-Output` instead of `Write-Host`. Explanation above.

Reply:
sure